### PR TITLE
Code modernization

### LINKED ISSUE #11: Modernize code
Use new features.

* [x] `string.format` instead of percents
* [x] modern properties syntax everywhere
* [x] remove object extensions in classes
* [x] super calls
* [x] `raise Foo() from` syntax
* [x] `exc_info` in logging calls instead of manually adding the exception string
* [x] modern ABC usage
* [x] avoid using double underscores
* [x] replace deprecated logging methods (e.g. `warn` -> `warning`)

### PR DESCRIPTION
**Pull request by languitar**
This will resolve #11. I suspect the commits have to be reviewed individually.

### CODE REVIEW

File: rsb/introspection/__init__.py [hunk 1]
```diff
@@ -544,10 +544,10 @@ def handle(event):
                             (p for p in self.__participants
                              if p.participant_id == participant_id),
                             None)
-                except Exception as e:
+                except Exception:
                     self.__logger.warn('Query event %s does not '
-                                       'properly address a participant: %s',
-                                       event, e)
+                                       'properly address a participant',
+                                       event, exc_info=True)
```
[the remark this answers is not in the document]
Comment (languitar) [reply rsb/introspection/__init__.py]: What do you think is the preferred style? `exe_info=True` ensures that the whole message and stack of the raised exception are printed.

[the remark this answers is not in the document]
Comment (languitar) [reply rsb/introspection/__init__.py]: Yes, in that sense it does. However, the presentation might be a bit clearer. Without that, the output looks like:

```
In [1]: import logging

In [2]: logging.basicConfig(level=logging.DEBUG)

In [3]: try:
   ...:     raise ValueError('This is the message')
   ...: except ValueError as e:
   ...:     logging.getLogger('test').warning('This is an error', exc_info=True)
   ...:     
WARNING:test:This is an error
Traceback (most recent call last):
  File "<ipython-input-3-089dfdfce727>", line 2, in <module>
    raise ValueError('This is the message')
ValueError: This is the message
```

[the remark this answers is not in the document]
Comment (scymtym) [reply rsb/introspection/__init__.py]: I see. I'm fine with either, then, and also with choosing different styles as appropriate for the individual logger calls.

[the remark this answers is not in the document]
Comment (languitar) [reply rsb/introspection/__init__.py]: I think my original rationale here was that the details in this case are not of immediate relevance to users seeing the message. Therefore I removed it here and not in other places.

**Comment by languitar** [comment]
So if everything is fine, would you want to merge this?

**Comment by scymtym** [comment]
> So if everything is fine, would you want to merge this?

Yeah, just wanted to give you a chance to reply to the most recent comment.

**Pull request merged**